### PR TITLE
EMD metric

### PR DESCRIPTION
**Pull request by anujanegi**

### CODE REVIEW

File: bsi_zoo/metrics.py [hunk 1]
```diff
@@ -68,7 +57,62 @@ def euclidean_distance(x, x_hat, *args, **kwargs):
...
+def jaccard_error(x, x_hat, *args, **kwargs):
+    # You can read more on the Jaccard score in Scikit-learn definition https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_score.html
+    return 1 - jaccard_score(x, x_hat, average="samples")
+
+
+def mse(x, x_hat, *args, **kwargs):
+    return mean_squared_error(x, x_hat)
+
+
+def EMD(x, x_hat, *args, **kwargs):
```
Comment (agramfort) [inline bsi_zoo/metrics.py:72]: if the other function is called mse call this one emd
```suggestion
def emd(x, x_hat, *args, **kwargs):
```

and actually it could be:

```suggestion
def emd(x, x_hat, *, orientation_type, subject, nnz, *args, **kwargs):
```

to be sure orientation_type, subject and nnz are passed

File: bsi_zoo/metrics.py [hunk 2]
```diff
@@ -68,7 +57,62 @@ def euclidean_distance(x, x_hat, *args, **kwargs):
...
+    subject = kwargs["subject"]
+    nnz = kwargs["nnz"]
+
+    if orientation_type == "fixed":
+        temp = np.linalg.norm(x, axis=1)
+        a = temp[np.nonzero(temp)]
+
+        temp = np.linalg.norm(x_hat, axis=1)
+        temp_ = np.partition(-temp, nnz)
+        b = -temp_[:nnz]  # get n(=nnz) max amplitudes
```
Comment (agramfort) [inline bsi_zoo/metrics.py:83]: you don't need to only select nnz sources here. The EMD can be between vectors of different sizes

**Review by agramfort — COMMENTED** [review]
can you add a test to make sure it works as expected?

File: bsi_zoo/metrics.py (hunk 2 above)
Comment (anujanegi) [reply bsi_zoo/metrics.py:83]: @agramfort  Yes, I agree.
I'm not sure how to get coordinates arranged in the same position order as x/x_hat.

Comment (anujanegi) [reply bsi_zoo/metrics.py:83]: which is why just I selected nnz vertices for now.

File: bsi_zoo/benchmark.py [hunk 3]
```diff
@@ -13,102 +16,125 @@ def __init__(
...
+        # (gamma_map, {"alpha": [0.99, 0.9, 0.8, 0.5]})
+    ]
+
+    df_results = []
+    for estimator, data_args_to_benchmark in estimators:
+        benchmark = Benchmark(
+            estimator, subject, metrics, data_args, data_args_to_benchmark
+        )
+        results = benchmark.run(nruns=5)
+        df_results.append(results)
```
[the remark this answers is not in the document]
Comment (agramfort) [reply bsi_zoo/benchmark.py:136]: it's not a problem. I think you want to benchmark all estimators in a consistent without making exceptions

**Comment by anujanegi** [comment]
fix #19
